File: src/main/Python/boj_step4/boj10810.py
import sys
cin = sys.stdin.readline

N, M = map(int, cin().split())
basket = [0] * N
for _ in range(M):
    i, j, k = map(int, cin().split())
    for index in range(i - 1, j):
        basket[index] = k
for ball in basket: print(ball, end=' ')

File: src/main/Cpp/boj_step4/boj10810.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int N, M;
    cin >> N >> M;
    vector<int> baskets(N);
    for (int _ = 0; _ < M; _++) {
        int i, j, k;
        cin >> i >> j >> k;
        for (int index = i - 1; index < j; index++)
            baskets[index] = k;
    }
    for (int & i : baskets) cout << i << ' ';
}